bindgen/ir/Enum.cpp

```diff
@@ -14,9 +14,11 @@ Enum::Enum(std::string name, std::string type,
 
 bool Enum::isAnonymous() const { return name.empty(); }
 
-std::shared_ptr<TypeDef> Enum::generateTypeDef() {
+std::shared_ptr<TypeDef>
+Enum::generateTypeDef(std::shared_ptr<Location> location) {
     assert(!isAnonymous());
-    return std::make_shared<TypeDef>("enum_" + name, shared_from_this());
+    return std::make_shared<TypeDef>("enum_" + name, shared_from_this(),
+                                     std::move(location));
```

Comment:
I'm not sure I understand why the `enum` doesn't have a `location` itself, which can be "reused" here.

Also, from `TypeDef::location` doc string:
> `nullptr` if type is located in main file or is generated.

this is generated, no?

Reply:
sorry, docstring is wrong, `location` is null only if it is generated